src/CommonRLInterface.jl

```diff
@@ -0,0 +1,61 @@
+import CommonRLInterface
+
+const CRL = CommonRLInterface
+
+#####
+# CommonRLEnv
+#####
+
+struct CommonRLEnv{T<:AbstractEnv} <: CRL.AbstractEnv
+    env::T
+end
+
+Base.convert(::Type{CRL.AbstractEnv}, env::AbstractEnv) = convert(CommonRLEnv, env)
+
+function Base.convert(::Type{CommonRLEnv}, env::AbstractEnv)
+    new_env = CommonRLEnv(env)
+    if ActionStyle(env) === FullActionSet()
+        @eval CRL.@provide CRL.valid_actions(x::typeof($new_env)) = get_legal_actions(x.env)
+        @eval CRL.@provide CRL.valid_action_mask(env::typeof($new_env)) = get_legal_actions_mask(env.env)
+    end
+    new_env
+end
```

Comment:
This is a case where I think it's best to manually implement `provided`

```suggestion
Base.convert(::Type{CommonRLEnv}, env::AbstractEnv) = CommonRLEnv(env)

CRL.provided(::typeof(CRL.valid_actions), env::CommonRLEnv) = ActionStyle(env.env) === FullActionSet()
CRL.valid_actions(x::typeof($new_env)) = get_legal_actions(x.env)

CRL.provided(::typeof(CRL.valid_action_mask), env::CommonRLEnv) = ActionStyle(env.env) === FullActionSet()
CRL.valid_action_mask(env::typeof($new_env)) = get_legal_actions_mask(env.env)
```

Reply:
Good suggestion!